tests/sphere_test.cpp

```diff
@@ -0,0 +1,100 @@
+#include "obj/objects.h"
+#include "gtest/gtest.h"
+
+using core::Ray;
+using core::RayHitData;
+using core::Vec3;
+using objects::Sphere;
+
+const float kSkipFreq = 1000;
+const Vec3 kVecZero(0, 0, 0);
+
+TEST(SphereCollisionTest, RayHitFromOutsideSphere) {
+  Sphere sphere(kVecZero, 1);
+  RayHitData hitData;
+
+  Ray alongYAxis(Vec3(0, -4, 0), Vec3(0, 1, 0));
+  ASSERT_TRUE(sphere.hitObject(alongYAxis, kSkipFreq, &hitData));
+  ASSERT_FLOAT_EQ(alongYAxis.origin().magnitude() - sphere.getRadius(),
+                  hitData.time);
+
+  Ray alongYAxisNoHit(Vec3(0, 4, 0), Vec3(0, 1, 0));
+  ASSERT_FALSE(sphere.hitObject(alongYAxisNoHit, kSkipFreq, &hitData))
+      << "hit at: " << hitData.collisionPoint();
+
+  Vec3 arbitraryVec3(12.123, 5.623, 13.235);
```

Comment:
Do you care if this is inside the shpere or not? if do you please add an assert that will verify that in case you change the sphere position.

Reply:
This should be updated now, with newets push ;)

Reply:
Still not clear if this is ok to be inside the sphere or not.

Reply:
now should be clear ;)